tokenize.c

```diff
@@ -78,16 +78,19 @@ char **tokenize(char *str, char delim)
 
 	tokens_counter = count_tokens(str, delim);
 
-	if (!tokens_counter)
+	if (tokens_counter == 0)
 		return (NULL);
 
 	str_arr = malloc((tokens_counter + 1) * sizeof(char *));
 	if (!str_arr)
-		return (NULL);
-
+		{
+			free(str);
```

Comment:
you don't need to free here, because if the output is `NULL`, the function using this will free also